### ISSUE TITLE
Link component when provided with a onClick function causes accessibility error

### ISSUE THREAD

**Issue by philearley** [Sage/carbon#2500]
When using the Link component with an onClick function it causes accessibility errors to be raised.

Code specific error in our linter: Anchor used as a button. Anchors are primarily expected to navigate. Use the button element instead.eslint(jsx-a11y/anchor-is-valid)

This is also reported by Axe

Looks like this has been raised previous here and closed: https://github.com/Sage/carbon/issues/1915

**Comment by jamime** [comment]
@philearley do you want the `Link` to look like a button or a hyperlink? What are you trying to achieve in the `onClick` handler?

**Comment by philearley** [comment]
We are using Link components with an Icon and conditionally providing an onClick function or a href. This sometimes causes a dialog to open or a redirect.

When you use the Link component, it accepts an onClick function in props.

When you provide this onClick function, this is generating eslint and accessibility warnings as it claims an onClick function should be used with a button.
